index.d.ts

```diff
@@ -0,0 +1,197 @@
+		alias: 'u',
+		default: 'rainbow'
+	}
+}
+	```
+	 */
+	readonly flags?: MinimistOptions;
+
+	/**
+	Description to show above the help text. Default: The package.json `"description"` property.
+
+	Set it to `false` to disable it altogether.
+	 */
+	readonly description?: string | false;
+
+	/**
+	The help text you want shown.
+
+	The input is reindented and starting/ending newlines are trimmed which means you can use a [template literal](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings) without having to care about using the correct amount of indent.
+
+	The description will be shown above your help text automatically.
+
+	Set it to `false` to disable it altogether.
+	 */
+	readonly help?: string | false;
+
+	/**
+	Set a custom version output. Default: The package.json `"version"` property.
+
+	Set it to `false` to disable it altogether.
+	 */
+	readonly version?: string | false;
+
+	/**
+	Automatically show the help text when the `--help` flag is present. Useful to set this value to `false` when a CLI manages child CLIs with their own help text.
+	 */
+	readonly autoHelp?: boolean;
+
+	/**
+	Automatically show the version text when the `--version` flag is present. Useful to set this value to `false` when a CLI manages child CLIs with their own version text.
+	 */
+	readonly autoVersion?: boolean;
+
+	/**
+	package.json as an `Object`. Default: Closest package.json upwards.
+
+	_You most likely don't need this option._
+	 */
+	readonly pkg?: {[key: string]: unknown};
+
+	/**
+	Custom arguments object.
+
+	@default process.argv.slice(2)
+	 */
+	readonly argv?: ReadonlyArray<string>;
+
+	/**
+	Infer the argument type.
+
+	By default, the argument `5` in `$ foo 5` becomes a string. Enabling this would infer it as a number.
+
+	@default false
+	 */
+	readonly inferType?: boolean;
+
+	/**
+	Value of `boolean` flags not defined in `argv`. If set to `undefined` the flags not defined in `argv` will be excluded from the result. The `default` value set in `boolean` flags take precedence over `booleanDefault`.
+
+	__Caution: Explicitly specifying undefined for `booleanDefault` has different meaning from omitting key itself.__
+
+	@example
+	```
+const cli = meow(`
+	Usage
+		$ foo
+
+	Options
+		--rainbow, -r  Include a rainbow
+		--unicorn, -u  Include a unicorn
+		--no-sparkles  Exclude sparkles
+
+	Examples
+		$ foo
+		🌈 unicorns✨🌈
+`, {
+	booleanDefault: undefined,
+	flags: {
+		rainbow: {
+			type: 'boolean',
+			default: true,
+			alias: 'r'
+		},
+			unicorn: {
+			type: 'boolean',
+			default: false,
+			alias: 'u'
+		},
+		cake: {
+			type: 'boolean',
+			alias: 'c'
+		},
+		sparkles: {
+			type: 'boolean',
+			default: true
+		}
+	}
+});
+
+//{
+//	flags: {
+//		rainbow: true,
+//		unicorn: false,
+//		sparkles: true
+//	},
+//	unnormalizedFlags: {
+//		rainbow: true,
+//		r: true,
+//		unicorn: false,
+//		u: false,
+//		sparkles: true
+//	},
+//	…
+//}
+	 ```
+	 */
+	readonly booleanDefault?: boolean | null;
```

Comment:
According to the docs this should support `undefined` too. It's unclear to me why `null` is mentioned in the docs though. Might be a mistake...

Reply:
It's already optional, so `undefined` is already included.

Reply:
Ah right... My mistake.

Reply:
Included it for clarity.